### ISSUE TITLE
Why `ParseCtx.compile` doesn't return a Result?

### ISSUE THREAD

**Issue by lquerel** [01mf02/jaq#200]
The implementation of `ParseCtx.compile(jaq_syn::Main)` is currently error-prone because, if internally the compilation fails, the `compile` method returns a default Filter (i.e., `.`) and the public field `ParseCtx.errs` is populated with the errors. This design makes the usage of `ParseCtx.compile()` a bit error-prone as it's not obvious that the field `errs` must be checked. So, my question is, why not return a `Result<Filter, Vec<Error>>` instead of `Filter` and get rid of the field `errs` in `ParseCtx`?

**Comment by 01mf02** [comment]
I agree with you totally. I'm currently working on jaq 2.0, and I plan to return `Result`s for functions like the one you mentioned. This will also cut down on other boilerplate as @Andrew15-5 suggested in #188.
